main/src/ca/uwaterloo/flix/language/phase/OccurrenceAnalyzer.scala

```diff
@@ -358,38 +360,61 @@ object OccurrenceAnalyzer {
   private def visitExps(exps: List[LiftedAst.Expression]): (List[OccurrenceAst.Expression], Map[Symbol.VarSym, Occur]) = {
     exps.foldRight((List[OccurrenceAst.Expression](), Map[Symbol.VarSym, OccurrenceAst.Occur]()))((exp, acc) => {
       val (e, o1) = visitExp(exp)
-      val o2 = combineAll(o1, acc._2)
+      val o2 = combineAllSeq(o1, acc._2)
       (e :: acc._1, o2)
     })
   }
 
+  /**
+   * Combines the 2 maps `m1` and `m2` of the type (Symbol -> Occur) into a single map of same type using the function `combineBranches`.
+   */
+  private def combineAllBranch(m1: Map[VarSym, Occur], m2: Map[VarSym, Occur]): Map[VarSym, Occur] = {
+    combineAll(m1, m2, combineBranch)
+  }
+
   /**
    * Combines the 2 maps `m1` and `m2` of the type (Symbol -> Occur) into a single map of same type using the function `combine`.
    */
-  private def combineAll(m1: Map[Symbol.VarSym, Occur], m2: Map[Symbol.VarSym, Occur]): Map[Symbol.VarSym, Occur] = {
-    (m1.keys ++ m2.keys).foldLeft[Map[Symbol.VarSym, Occur]](Map.empty) {
-      case (acc, k) => acc + (k -> combineOpt(m1.get(k), m2.get(k)))
-    }
+  private def combineAllSeq(m1: Map[VarSym, Occur], m2: Map[VarSym, Occur]): Map[VarSym, Occur] = {
+    combineAll(m1, m2, combineSeq)
   }
 
-  /*
-  * Combines `o1` and `o2` if both contain a value, else return the option containing a value.
+  /**
+   * Combines the 2 maps `m1` and `m2` of the type (Symbol -> Occur) into a single map of same type using the argument `combine`.
    */
-  private def combineOpt(o1: Option[Occur], o2: Option[Occur]): Occur = {
-    (o1, o2) match {
-      case (None, None) => throw InternalCompilerException(s"Unexpected options.")
-      case (None, Some(o2)) => o2
-      case (Some(o1), None) => o1
-      case (Some(o1), Some(o2)) => combine(o1, o2)
+  private def combineAll(m1: Map[VarSym, Occur], m2: Map[VarSym, Occur], combine: (Occur, Occur) => Occur): Map[VarSym, Occur] = {
+    (m1.keys ++ m2.keys).foldLeft[Map[VarSym, Occur]](Map.empty) {
+      case (acc, k) =>
+        val occur = (m1.get(k), m2.get(k)) match {
```

Comment:
What about simplifying this to: `combine(m1.getWithDefault(k, Dead), m2.getWithDefault(....)`?